### PR TITLE
Java Service Wrapper 3.5.46 (new formula)

### PR DESCRIPTION
**Pull request by VonUniGE**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Java Service Wrapper 3.5.46 is the fist Apple silicon compatible version. It can be used to start Sonarqube on Apple Silicon as discussed in https://github.com/Homebrew/homebrew-core/pull/88380

### CODE REVIEW

File: Formula/java-service-wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class JavaServiceWrapper < Formula
+  desc "Simplify the deployment, launch and monitoring of Java applications"
+  homepage "https://wrapper.tanukisoftware.com/"
+  url "https://downloads.sourceforge.net/project/wrapper/wrapper_src/Wrapper_3.5.46_20210903/wrapper_3.5.46_src.tar.gz"
+  sha256 "82e1d0c85488d1389d02e3abe3359a7f759119e356e3e3abd6c6d67615ae5ad8"
+  license "GPL-2.0-only"
+
+  depends_on "openjdk@11" => :build
```
Comment (carlocab) [inline Formula/java-service-wrapper.rb:8]:
```suggestion
  depends_on "ant" => :build
  depends_on "openjdk@11" => :build
```
This seems to be needed.

File: Formula/java-service-wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def install
+    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_prefix
+    system "ant", "-Dbits=64", "-Djavac.target.version=1.6"
+    libexec.install "lib", "bin", "src/bin" => "scripts"
+  end
+
+  test do
+    shell_output("#{libexec}/bin/testwrapper status", 1)
```
Comment (carlocab) [inline Formula/java-service-wrapper.rb:18]: Is there output we can `assert_match` here?

File: Formula/java-service-wrapper.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "https://wrapper.tanukisoftware.com/"
+  url "https://downloads.sourceforge.net/project/wrapper/wrapper_src/Wrapper_3.5.46_20210903/wrapper_3.5.46_src.tar.gz"
+  sha256 "82e1d0c85488d1389d02e3abe3359a7f759119e356e3e3abd6c6d67615ae5ad8"
+  license "GPL-2.0-only"
+
+  depends_on "openjdk@11" => :build
+
+  def install
+    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_prefix
+    system "ant", "-Dbits=64", "-Djavac.target.version=1.6"
```
Comment (carlocab) [inline Formula/java-service-wrapper.rb:13]: A comment explaining the `1.6` here would be good.

**Review by carlocab — COMMENTED** [review]
Nice work here. A few comments.

File: Formula/java-service-wrapper.rb (hunk 1 above)
Comment (carlocab) [reply Formula/java-service-wrapper.rb:8]: Ah, I see you just pushed this.

File: Formula/java-service-wrapper.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class JavaServiceWrapper < Formula
+  desc "Simplify the deployment, launch and monitoring of Java applications"
+  homepage "https://wrapper.tanukisoftware.com/"
+  url "https://downloads.sourceforge.net/project/wrapper/wrapper_src/Wrapper_3.5.46_20210903/wrapper_3.5.46_src.tar.gz"
+  sha256 "82e1d0c85488d1389d02e3abe3359a7f759119e356e3e3abd6c6d67615ae5ad8"
+  license "GPL-2.0-only"
+
+  depends_on "openjdk@11" => :build
+  depends_on "ant" => :build
```
Comment (carlocab) [inline Formula/java-service-wrapper.rb:9]: Dependencies should be listed alphabetically.

File: Formula/java-service-wrapper.rb (hunk 2 above)
Comment (VonUniGE) [reply Formula/java-service-wrapper.rb]: Yes, done.

File: Formula/java-service-wrapper.rb (hunk 3 above)
Comment (VonUniGE) [reply Formula/java-service-wrapper.rb:14]: Ok, done.

File: Formula/java-service-wrapper.rb (hunk 4 above)
Comment (VonUniGE) [reply Formula/java-service-wrapper.rb:8]: Ok, done.

**Comment by carlocab** [comment]
Linux failed with:
```
       [exec] testsuite.c:21:29: fatal error: CUnit/Automated.h: No such file or directory
       [exec] compilation terminated.
       [exec] test_example.c:15:25: fatal error: CUnit/Basic.h: No such file or directory
       [exec] compilation terminated.
       [exec] test_javaadditionalparam.c:15:25: fatal error: CUnit/Basic.h: No such file or directory
       [exec] compilation terminated.
       [exec] test_hashmap.c:15:25: fatal error: CUnit/Basic.h: No such file or directory
       [exec] compilation terminated.
       [exec] test_filter.c:15:25: fatal error: CUnit/Basic.h: No such file or directory
       [exec] compilation terminated.
       [exec] wrapper.c:52:25: fatal error: CUnit/Basic.h: No such file or directory
       [exec] compilation terminated.
       [exec] make: *** [testsuite] Error 1
       [exec] make: *** Waiting for unfinished jobs....
       [exec] Makefile-linux-x86-64.make:44: recipe for target 'testsuite' failed
```

File: Formula/java-service-wrapper.rb (hunk 3 above)
Comment (iMichka) [reply Formula/java-service-wrapper.rb:14]: Is there a potential risk something breaks as upstream shipped with version 1.4 and probably never tested the build with 1.6? I am a little bit worried of hard-to-debug compile / runtime exceptions.

**Comment by iMichka** [comment]
Yes, you need to add `depends_on "cunit" => :build` for Linux only.

File: Formula/java-service-wrapper.rb (hunk 3 above)
Comment (VonUniGE) [reply Formula/java-service-wrapper.rb:17]: In my opinion, the risk is limited. It all depends on your context, if this is for a development environment the risk is acceptable, if this is for a mission critical application, you should buy a [server license](https://wrapper.tanukisoftware.com/doc/english/licenseServer.html) and use the official build.

Maybe we can add a sentence in the caveat about this ?

Comment (carlocab) [reply Formula/java-service-wrapper.rb:17]: There's no need; that caveat would apply to everything we ship.

**Review by carlocab — APPROVED** [review]
Thanks, @VonUniGE!

**Pull request closed without merging**